packages/components/src/internal/components/samples/SampleCreationTypeModal.tsx

```diff
@@ -0,0 +1,144 @@
+    options: Array<SampleCreationTypeModel>,
+    parentCount: number;
+    showIcons: boolean;
+    onCancel: () => void;
+    onSubmit: (creationType: SampleCreationType, numPerParent?: number) => void;
+}
+
+interface State {
+    numPerParent: number;
+    creationType: SampleCreationType;
+    submitting: boolean;
+}
+
+export class SampleCreationTypeModal extends React.PureComponent<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+
+        props.options.find(option => option.selected);
+        this.state = {
+            creationType: props.options.find(option => option.selected)?.type || props.options[0].type,
+            numPerParent: 1,
+            submitting: false,
+        }
+    }
+
+    onCancel = () => {
+        this.props.onCancel();
+    };
+
+    onChange = event => {
+        const { name, value } = event.target;
+        this.setState({ [name]: value } as State);
+    };
+
+    renderNumPerParent() : React.ReactNode {
+        const { parentCount, options } = this.props;
+        const { creationType, numPerParent } = this.state;
+
+        const selectedOption = options.find(option => option.type === creationType);
+        return (
+            <>
+                {this.shouldDisplayOptions() && <hr/>}
+                <div>
+                    <label className="creation-type-modal-label">{selectedOption.quantityLabel}</label>
+                    <label className="creation-type-modal-label">
+                        <FormControl
+                            className="creation-per-parent-select"
+                            min={1}
+                            max={MAX_EDITABLE_GRID_ROWS/parentCount}
+                            step={1}
+                            name={"numPerParent"}
+                            onChange={this.onChange}
+                            type="number"
+                            value={numPerParent}
+                        />
+                    </label>
+                </div>
+                Sample details and quantities can be modified on the grid.
+            </>
+        )
+    }
+
+    onConfirm = () => {
+        this.props.onSubmit(this.state.creationType, this.state.numPerParent);
+    }
+
+    shouldDisplayOptions() : boolean {
+        return this.getOptionsToDisplay().length > 1;
+    }
+
+    shouldRenderOption(option: SampleCreationTypeModel) : boolean {
+        return this.props.parentCount >= option.minParentsPerSample;
+    }
+
+    getOptionsToDisplay() : Array<SampleCreationTypeModel> {
+        return this.props.options.filter(option => this.shouldRenderOption(option));
+    }
+
+    renderOptions() : Array<React.ReactNode> {
+        const { showIcons, options } = this.props;
+        const displayOptions = this.getOptionsToDisplay();
+        if (displayOptions.length < 2)
+            return null;
+
+        let optionSet = [];
+        options.forEach((option, i) => {
+            if (this.shouldRenderOption(option)) {
```

Comment:
looks like the `displayOptions` const above already filtered for this, any reason not to just use that here for the `forEach`?